specs/interop/predeploys.md

```diff
@@ -237,18 +237,34 @@ getters.
 |----------|----------------------------------------------|
 | Address  | `0x4200000000000000000000000000000000000015` |
 
-The `L1Block` contract is updated to include the set of allowed chains. The L1 Attributes transaction
-sets the set of allowed chains. The `L1Block` contract MUST provide a public getter to check if a particular
-chain is in the dependency set called `isInDependencySet(uint256)`. This function MUST return true when
-the chain's chain id is passed in as an argument.
+The `L1Block` contract MUST include method `setConfig(ConfigType, bytes)` for setting static values that are used
+by the system. The `ConfigType` enum is defined as follows:
 
-The `setL1BlockValuesInterop()` function MUST be called on every block after the interop upgrade block.
+```solidity
+enum ConfigType {
+    GAS_PAYING_TOKEN,
+    BATCHER_HASH,
+    ADD_DEPENDENCY,
+    REMOVE_DEPENDENCY
+}
+```
+
+The second argument of `setConfig` is the encoded value being set.
+Calls to `setConfig` MUST originate from `SystemConfig` and are forwarded to `L1Block` by `OptimismPortal`.
+
+`L1Block` is updated to include the set of allowed chains. These chains are added and removed through `setConfig`
+calls with `ADD_DEPENDENCY` or `REMOVE_DEPENDENCY`, respectively.
+
+`L1Block` MUST provide a public getter to check if a particular chain is in the dependency set called
+`isInDependencySet(uint256)`. This function MUST return true when the chain's chain id is passed in as an argument.
+
+The `setL1BlockValuesHolocene()` function MUST be called on every block after the interop upgrade block.
 The interop upgrade block itself MUST include a call to `setL1BlockValuesEcotone`.
 
 ### L1Attributes
 
-The L1 Atrributes transaction is updated to include the dependency set. Since the dependency set is dynamically sized,
-a `uint8` "interopSetSize" parameter prefixes tightly packed `uint256` values that represent each chain id.
+The L1 Attributes transaction is updated to only include dynamic values. Static values, such as `batcherHash` and
+the interop dependency set, are set through the `setConfig` method.
 
 | Input arg         | Type                     | Calldata bytes          | Segment |
 |-------------------|--------------------------|-------------------------|---------|
```

Comment:
I am noticing that the fee scalars are also static config, we could simplify this more by adding enum fields for them

Reply:
This would effectively replace https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/diff-based-attributes-tx.md

The node implementation section is important to be mindful of as it would also apply to this change

Reply:
good idea!

Reply:
If we also use the static path for the fee scalars, then we will need to guarantee that we can always update them, meaning that we would need to stage the upgrades such that the L2 hardforks and adds the new path for setting them but the L1 contracts don't call them yet, then we would need to update the L1 contracts to be able to call them, then we would need to update the L2 contract to only use the static setter path

Reply:
That is too big of a decision to make in this PR, it would require a design doc and presenting it at the design review session, so i recommend we just stick with custom gas token + interop related features for now and revisit the idea later

Reply:
I think the best next steps are to not modify the L1 attributes tx at all from ecotone and still follow the old flow of doing the updates for the old static attributes but have the `L1Block` contract ready with the new setters for each of the static fields. Then we can have a smooth transition and deprecate the old path. This removes a synchronous upgrade requirement

Reply:
since L1 Attributes tx is not modified, dropping the section in about it